### PR TITLE
Remove beta mark from APIs introduced for JEP-202

### PR DESCRIPTION
**Pull request by jglick**
@oleg-nenashev pointed out in https://github.com/jenkinsci/jep/pull/280 that for the JEP to be considered final we would expect these APIs to become unrestricted. They have long been in use in plugin releases.

### Proposed changelog entries

* Pluggable Artifact Storage: Make the `VirtualFile` API generally available to plugin developers.
  * JEP-202: https://github.com/jenkinsci/jep/tree/master/jep/202

### Proposed upgrade guidelines

N/A

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]

File: core/src/main/java/jenkins/util/VirtualFile.java
Comment (jglick) [inline core/src/main/java/jenkins/util/VirtualFile.java:69]: BTW @Wadeck there were some APIs you introduced later with a review comment saying that they should be unrestricted in a weekly; did you forget?

**Review by res0nance — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
Fine with me. We have 2+ reference implementations, including one created by external developers (Azure artifact storage)

**Comment by oleg-nenashev** [comment]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

File: core/src/main/java/jenkins/util/VirtualFile.java
Comment (Wadeck) [reply core/src/main/java/jenkins/util/VirtualFile.java:69]: Actually they are not yet ready / finalized

**Review by timja — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
restarted CI. Will merge once/if it passes

**Pull request merged**
